### PR TITLE
nRF5340 support in ESB library

### PR DESCRIPTION
**Pull request by too1**
Changes:
- Support added for the nRF5340 network core
- nRF52832 rev 1 workarounds removed
- nrfx_ppi and nrfx_dppi support added

### CODE REVIEW

File: subsys/esb/esb.c [hunk 1]
```diff
@@ -547,25 +482,37 @@ static void sys_timer_init(void)
...
+	nrfx_ppi_channel_alloc(&ppi_ch_timer_compare1_radio_txen);
+
+	nrfx_ppi_channel_assign(ppi_ch_radio_ready_timer_start, 
+		(uint32_t)&NRF_RADIO->EVENTS_READY, (uint32_t)&ESB_SYS_TIMER->TASKS_START);
+	nrfx_ppi_channel_assign(ppi_ch_radio_address_timer_stop, 
+		(uint32_t)&NRF_RADIO->EVENTS_ADDRESS, (uint32_t)&ESB_SYS_TIMER->TASKS_SHUTDOWN);
+	nrfx_ppi_channel_assign(ppi_ch_timer_compare0_radio_disable, 
+		(uint32_t)&ESB_SYS_TIMER->EVENTS_COMPARE[0], (uint32_t)&NRF_RADIO->TASKS_DISABLE);
+	nrfx_ppi_channel_assign(ppi_ch_timer_compare1_radio_txen, 
+		(uint32_t)&ESB_SYS_TIMER->EVENTS_COMPARE[1], (uint32_t)&NRF_RADIO->TASKS_TXEN);
```
Comment (kapi-no) [inline subsys/esb/esb.c:512]: Have you considered using nrfx abstraction over PPI / DPPI?

https://github.com/nrfconnect/sdk-hal_nordic/blob/master/nrfx/helpers/nrfx_gppi.h

If API is flexible enough to cover ESB logic, then we will have uniform code for both nRF52 and nRF53. You can see how it is used in the DTM example:

https://github.com/nrfconnect/sdk-nrf/blob/master/samples/bluetooth/direct_test_mode/src/dtm.c

**Review by lemrey — COMMENTED** [review]
Ah, good! ESB needed some love 🥇
You'll need to reword the commits to please gitlint :)

File: subsys/esb/esb.c (hunk 1 above)
Comment (too1) [reply subsys/esb/esb.c:512]: I use GPPI where possible, but channel allocation does not seem to be covered by GPPI for some reason. 
I discussed this with Krzysztof C, and he confirmed. 

The DTM example uses a hard coded PPI channel, so channel allocation is not used.

**Review by grochu — CHANGES_REQUESTED** [review]
Added a few comments to the code. Additionally, please update the commit messages to comply with the NCS style: use "_-s_" option, and add "_esb:_" headings.

**Review by grochu — APPROVED** [review]
Looks good, the PPI usage was discussed offline.

**Review by lemrey — APPROVED** [review]
Thank you!

**Review by lemrey — CHANGES_REQUESTED** [review]
We have some system tests so we can try to run those. Hang on.

**Review by lemrey — APPROVED** [review]

**Comment by lemrey** [comment]
@too1 I have rebased this PR and updated the changelog, could you please have a look to make sure it contains the correct information? @umapraseeda Could you please review my changes to the changelog?

**Comment by grochu** [comment]
I tested the sample on the nRF5340 DK against Graviton and it works OK.

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -22,9 +22,24 @@ nRF5
...
 The following changes are relevant for the nRF52 and nRF53 Series.
 
+nRF52832 SoC
+------------
+
+* Removed:
+
+  * ``esb`` subystem - Removed support for older SoC revision workarounds.
+    Removed support for nRF52832 revision 1 workarounds for Errata 102, Errata 106, Errata 107.
+    Removed support for nRF52832 revision 2 workarounds for Errata 143.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-latest.rst:32]:
```suggestion
  * ``esb`` subystem - Removed support for older SoC revision workarounds.

    * Removed support for nRF52832 revision 1 workarounds for Errata 102, Errata 106, and Errata 107.
    * Removed support for nRF52832 revision 2 workarounds for Errata 143.
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -22,9 +22,24 @@ nRF5
...
+
+
 nRF5340 SoC
 -----------
 
+* Added:
+
+  * ``esb`` subsystem - Added support
+    Added support for nRF5340 SoC (CPUNET) in ESB subsystem.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-latest.rst:41]:
```suggestion
  * ``esb`` subsystem - Added support for nRF5340 SoC (CPUNET) in ESB subsystem.
```

**Review by umapraseeda — APPROVED** [review]

**Review by lemrey — APPROVED** [review]
@grochu kindly helped test this on the DK, the docs are updated.
I think we are ready now! 👍🏻

**Comment by lemrey** [comment]
or actually we should update the `yaml` files :)

**Review by lemrey — APPROVED** [review]
Looking good now

**Review by tejlmand — APPROVED** [review]

**Pull request merged**